### PR TITLE
Test

### PR DESCRIPTION
**Pull request by xueyang2018**
update test

### CODE REVIEW

File: src/pages/visualSettings.page.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      id =>
+        id
+          .$('.dropdown-item-action')
+          .getText()
+          .toUpperCase() !== currentlySkinValue
+    );
+    skinItem.click();
+    this.applyBtn.click();
+
+    this.header.waitForVisible();
```
Comment (Yosure) [inline src/pages/visualSettings.page.js]: 这句应该起不到作用

File: src/pages/visualSettings.page.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    expect(this.logoBar.getText()).toContain('Sprint');
```
Comment (Yosure) [inline src/pages/visualSettings.page.js]: 这个是在验证什么？

File: src/pages/visualSettings.page.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    this.clearVisualSettings(currentlySkinValue);
+  }
+
+  clearVisualSettings(resetTheme) {
```
Comment (Yosure) [inline src/pages/visualSettings.page.js]: resetSkin

File: test/features/administration.feature [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Feature: Administration Tab
+
+  Scenario: Manage Visual Settings
+    Given I logged in as "adminOne"
+    When I select a organization
+    And I select "Administration" menu
+    And I select Manage Visual Settings
+    And I can manage Visual Settings
+
```
Comment (Yosure) [inline test/features/administration.feature]: 你做这些操作的目的和结果Then呢

File: wdio.conf.js [hunk 5]
```diff
@@ -65,21 +63,18 @@ exports.config = {
   coloredLogs: true,
   deprecationWarnings: true,
   screenshotPath: './results/screenshots',
-  waitforTimeout: 50000,
+  waitforTimeout: 80000,
```
Comment (Yosure) [inline wdio.conf.js]: 时间是你改的？如果是，改回原来状态。

File: wdio.test.conf.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ const { config } = require('./wdio.conf');
  */
 
 config.baseUrl = 'https://env2.smartworks.com/';
-config.specs = [ './test/features/cart.feature' ];
+config.specs = [ './test/features/administration.feature' ];
```
Comment (Yosure) [inline wdio.test.conf.js]: 改回到原来状态

File: src/pages/visualSettings.page.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+  get skinBtn() {
+    return $('#SelectSkinButton');
+  }
+
+  get skinList() {
+    return $('#SelectSkinList');
+  }
+
+  get itemsList() {
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: has updated as "skinItems"

File: src/pages/visualSettings.page.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    super();
+    this.href = '/Administration/ManageVisualSettings';
+  }
+
+  trait() {
+    super.trait(this.href);
+    expect(this.header.getText()).toBe('Manage Visual Settings');
+  }
+
+  setVisualSettings() {
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: Has updated as "manageVisualSettings"

File: src/pages/organizationHome.page.js [hunk 9]
```diff
@@ -41,16 +66,69 @@ class OrganizationHomePage extends Page {
...
+      .$('.stack-menu-item-link')
+      .getAttribute('data-nextmenu')
+      .replace(/[^0-9]/gi, '');
     selectSubCatalog.click();
+
     const [, , selectAdditionalCatalog] = this.browseAdditionalCatalogs;
     selectAdditionalCatalog.click();
   }
 
+  clickSettingIcon(menu) {
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/organizationHome.page.js]: Has updated as "findAdminMenu"

File: src/pages/organizationHome.page.js [hunk 10]
```diff
@@ -41,16 +66,69 @@ class OrganizationHomePage extends Page {
...
+    this.settingIcon.click();
+    const selectedSettingMenu = this.adminListItemsForConditionJudgment.find(
+      item =>
+        item
+          .$('.inner-link-text')
+          .getText()
+          .replace(/\s*$/g, '') === menu
+    );
+
+    if (!selectedSettingMenu) {
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/organizationHome.page.js]: No, it has been judged. Please see "if"

[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/organizationHome.page.js]: This one need to update as "while"

File: src/pages/organizationHome.page.js [hunk 11]
```diff
@@ -41,16 +66,69 @@ class OrganizationHomePage extends Page {
...
+    }
+  }
+
+  selectSettingMenu(menu) {
+    this.clickSettingIcon(menu);
+    const selectedSettingMenu = this.adminListItems.find(
+      item => item.getText().replace(/\s*$/g, '') === menu
+    );
+    selectedSettingMenu.click();
+    this.selectedSettingMenuList.$('.dropdown-menu').waitForVisible();
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/organizationHome.page.js]: has discussed, it is ok

File: src/pages/visualSettings.page.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: has updated as "manageVisualSettings"

File: src/pages/visualSettings.page.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    this.skinBtn.waitForVisible();
+    const currentlySkinValue = this.skinBtn
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: has updated as "currentlySkinName"

File: src/pages/visualSettings.page.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+      .$('#SelectSkinButtonText')
+      .getText();
+    this.skinBtn.click();
+    this.skinList.waitForVisible();
+
+    const skinItem = this.itemsList.find(
+      id =>
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: has updated as "item"

File: src/pages/visualSettings.page.js [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+
+  setVisualSettings() {
+    this.skinBtn.waitForVisible();
+    const currentlySkinValue = this.skinBtn
+      .$('#SelectSkinButtonText')
+      .getText();
+    this.skinBtn.click();
+    this.skinList.waitForVisible();
+
+    const skinItem = this.itemsList.find(
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: has updated as "newSkin"

File: src/pages/visualSettings.page.js (hunk 1 above)
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: this one need to update as "judge button is enable or not"

File: src/pages/visualSettings.page.js (hunk 2 above)
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: the same with above

File: test/features/administration.feature (hunk 4 above)
Comment (xueyang2018) [reply test/features/administration.feature]: has updated as "then"

File: wdio.conf.js (hunk 5 above)
Comment (xueyang2018) [reply wdio.conf.js]: keep consistent with original

File: wdio.test.conf.js (hunk 6 above)
Comment (xueyang2018) [reply wdio.test.conf.js]: keep consistent with original

File: src/pages/visualSettings.page.js (hunk 3 above)
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: has updated as "resetSkin"

File: src/pages/visualSettings.page.js [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+    expect(this.header.getText()).toBe('Manage Visual Settings');
+  }
+
+  setVisualSettings() {
+    this.skinBtn.waitForVisible();
+    const currentlySkinValue = this.skinBtn
+      .$('#SelectSkinButtonText')
+      .getText();
+    this.skinBtn.click();
+    this.skinList.waitForVisible();
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/visualSettings.page.js]: discuss and it is ok.

File: src/pages/organizationHome.page.js [hunk 16]
```diff
@@ -41,16 +66,69 @@ class OrganizationHomePage extends Page {
...
+  }
+
+  visualSettingsMgt() {
```
[the remark this answers is not in the document]
Comment (xueyang2018) [reply src/pages/organizationHome.page.js]: has updated as parameter

File: src/pages/organizationHome.page.js [hunk 17]
```diff
@@ -41,16 +66,73 @@ class OrganizationHomePage extends Page {
...
+    this.settingIcon.click();
+    let selectedSettingMenu = this.adminListItemsForConditionJudgment.find(
+      item =>
+        item
+          .$('.inner-link-text')
+          .getText()
+          .replace(/\s*$/g, '') === menu
+    );
+
+    // console.log(selectedSettingMenu);
```
Comment (Yosure) [inline src/pages/organizationHome.page.js]: 去掉

File: src/pages/organizationHome.page.js [hunk 18]
```diff
@@ -41,16 +66,73 @@ class OrganizationHomePage extends Page {
...
+      this.adminMenuArrowRight.click();
+      selectedSettingMenu = this.adminListItemsForConditionJudgment.find(
+        item =>
+          item
+            .$('.inner-link-text')
+            .getText()
+            .replace(/\s*$/g, '') === menu
+      );
+    }
+    // console.log(selectedSettingMenu);
```
Comment (Yosure) [inline src/pages/organizationHome.page.js]: 去掉

File: src/pages/organizationHome.page.js [hunk 19]
```diff
@@ -41,16 +66,73 @@ class OrganizationHomePage extends Page {
...
+  }
+
+  selectSettingMenu(menu) {
+    this.findAdminMenu(menu);
+    const selectedSettingMenu = this.adminListItems.find(
+      item => item.getText().replace(/\s*$/g, '') === menu
+    );
+    // console.log(selectedSettingMenu);
```
Comment (Yosure) [inline src/pages/organizationHome.page.js]: 去掉

**Comment by xueyang2018** [comment]
Ok , 注释本想提交后，没有问题了，再去掉的。我再修改一下吧

**Pull request merged**